app/templates/components/author-assertions.hbs

```diff
@@ -37,7 +37,11 @@
                                 analyticsName=(t 'components.author-assertions.public_data.title')
                             }}
                         {{else if (eq preprint.hasDataLinks 'no')}}
-                            {{~t 'components.author-assertions.public_data.no'~}}
+                            {{#if preprint.whyNoData}}
+                                {{preprint.whyNoData}}
+                            {{else}}
+                                {{~t 'components.author-assertions.public_data.no'~}}
+                            {{/if}}
```

Comment:
Same needs to be done [below](https://github.com/CenterForOpenScience/ember-osf-preprints/pull/705/files#diff-02aab108141199396345ae5399e10111R66) for `whyNoPrereg`?